### PR TITLE
added guts of custom elements semantics section

### PR DESCRIPTION
**Pull request by stevefaulkner**
not finished but far enough along to be reviewed

### CODE REVIEW

File: spec/custom/index.html [hunk 1]
```diff
@@ -754,6 +756,52 @@ <h2 id="es6">Custom Elements and ECMAScript 6</h2>
...
+  </ul>
+<h3>Custom Tag Semantics</h3>
+<p>By default a <a href="#dfn-custom-tag">custom tag</a> has no special meaning at all. It <a href="http://www.w3.org/TR/html/dom.html#represents">represents</a> its children.</p>
+<p>For example a custom tag, could be named <em>taco-button</em> but it does not express the semantics of a HTML <a href="http://www.w3.org/TR/html/forms.html#the-button-element"><code>button</code></a> element simply due to its name. As instaniated a custom tag conveys a similar amount of semantics as a HTML <a href="http://www.w3.org/TR/html/grouping-content.html#the-div-element"><code>div</code></a> or <code><a href="http://www.w3.org/TR/html/text-level-semantics.html#the-span-element">span</a></code> element. The addition of visual styling and scripted events to the <em>taco-button</em> could provide hints as to the semantics and expected interaction behaviours of the custom element for <em>some</em> users, but for the semantics to be formally expressed, developers must convey the role, states and properties of the element to browser APIs.</p>
+<pre>&lt;!-- taco-button represents a span with a fancy name -->
```
Comment (dglazkov) [inline spec/custom/index.html]: Brace with `<code class="prettyprint">` here and on. See example above (look for "amazement" in text)

File: spec/custom/index.html [hunk 2]
```diff
@@ -754,6 +756,52 @@ <h2 id="es6">Custom Elements and ECMAScript 6</h2>
...
+onkeypress=&quot;if(event.keyCode==32||event.keyCode==13){alert('tasty eh?');};&quot;&gt;Eat Me&lt;/taco-button&gt;</pre>
+<h3>Type Extension Semantics </h3>
+<p>By default a <a href="#dfn-type-extension">type extension</a> inherits the semantics of the element type it extends.</p>
+<p>For example a type extension, could extend the HTML  <a href="http://www.w3.org/TR/html/forms.html#the-button-element"><code>button</code></a> element. As instaniated it would inherit the <code>button</code> element's name, role, states and properties, built in focus and keyboard interaction behaviours. </p>
+<pre>&lt;!-- taco-button represents a button with an accesssible name of &quot;Eat Me!&quot; -->
+&lt;button <mark>is=&quot;taco-button&quot;</mark>&gt;Eat Me!&lt;/taco-button&gt;
+</pre>
+<p>To provide the desired <em>taco-button</em> feature, all that is required is the addition of an event handler. The rest of the semantics and interaction behaviour are provided by the browser as part of its implementation of the <a href="http://www.w3.org/TR/html/forms.html#the-button-element"><code>button</code></a> element.</p>
```
Comment (dglazkov) [inline spec/custom/index.html:801]: Is this informative content? Maybe mark it as such? I am not sure, this is necessary. The event handlers could be added by the lifecycle callbacks imperatively, or maybe even not used at all.

**Comment by stevefaulkner** [comment]
added pretty styles and note about event handlers and styles for note/example code

File: spec/custom/index.html (hunk 1 above)
Comment (stevefaulkner) [inline spec/custom/index.html]: done

File: spec/custom/index.html (hunk 2 above)
Comment (stevefaulkner) [inline spec/custom/index.html:801]: added note

**Comment by dglazkov** [comment]
LGTM.

**Pull request merged**